modules/bioinspired/src/retina_ocl.cpp

```diff
@@ -48,70 +48,37 @@
 #include <iostream>
 #include <sstream>
 
-#ifdef HAVE_OPENCV_OCL
+#ifdef HAVE_OPENCL
 
-#include "opencl_kernels.hpp"
+#include "opencl_kernels_bioinspired.hpp"
 
 #define NOT_IMPLEMENTED CV_Error(cv::Error::StsNotImplemented, "Not implemented")
 
+namespace
+{
+    template <typename T, size_t N>
+    inline int sizeOfArray(const T(&)[N])
+    {
+        return (int)N;
+    }
+
+    inline void ensureSizeIsEnough(int rows, int cols, int type, cv::UMat &m)
+    {
+        m.create(rows, cols, type, m.usageFlags);
+    }
+}
+
 namespace cv
 {
 static ocl::ProgramEntry retina_kernel = ocl::bioinspired::retina_kernel;
+static ocl::ProgramSource retina_kernel_source(retina_kernel.programStr);
```

Comment:
These 2 lines should be defined in `opencl_kernels_bioinspired.hpp` and `opencl_kernels_bioinspired.cpp` (located in `<build_dir>/modules/bioinspired`). No need to re-declare them.